CMakeLists.txt

```diff
@@ -1,5 +1,11 @@
 CMAKE_MINIMUM_REQUIRED(VERSION 2.8.7 FATAL_ERROR)
 
+if (NOT ${CMAKE_BINARY_DIR} STREQUAL ${CMAKE_SOURCE_DIR})
+  message(FATAL_ERROR
+    "HHVM requires you to build in-source.  Please run 'cmake .' "
```

Comment:
We don't, not entirely anyways.
I had to make a lot of changes to make it work for Windows, but it does work. We have conflicting `config.h` headers in our dependencies, but your work with setting up the individual projects correctly in third-party should be able to solve that by handling the include directory order correctly.

We don't respect it everywhere either, but it's better than it would be otherwise.

All of my work on Windows has been with out-of-tree builds.